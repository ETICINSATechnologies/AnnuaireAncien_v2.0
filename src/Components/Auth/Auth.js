class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
