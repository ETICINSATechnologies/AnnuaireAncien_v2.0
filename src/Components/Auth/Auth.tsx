class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
