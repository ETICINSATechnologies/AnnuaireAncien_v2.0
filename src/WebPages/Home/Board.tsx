import React, {Component} from 'react';
import defaultMan from '../../Images/default_man.svg'
import defaultWoman from '../../Images/default_woman.svg'
import {Member} from "../../Model/Member";

interface BoardProps {
    members: Member[]
    position: string
}

interface BoardState {
    divHeight: number
}

class Board extends Component<BoardProps, BoardState> {
    state = {
        divHeight: 0
    };
    boardRef: any;

    constructor(props: BoardProps) {
        super(props);
    }

    setDivHeight = () => {
        if (this.boardRef) {
            let divHeight = window.innerHeight - this.boardRef.getBoundingClientRect().y;
            this.setState({divHeight: divHeight});
        }
    };

    componentDidMount() {
        this.setDivHeight();
        window.addEventListener('resize', this.setDivHeight)
    }

    render() {
        const rowsNb = Math.max(this.props.members.length, 5);
        const rowSize = Math.min(this.state.divHeight / rowsNb, 250);

        let columnHeight;
        let boardStyle;

        if (window.innerHeight < window.innerWidth) {
            columnHeight = `${1.2 * rowSize}px`;
            boardStyle = {
                gridTemplateRows: `repeat(${rowsNb}, ${rowSize}px)`
            }
        }
        else {
            columnHeight = '12vw';
            boardStyle = {
                gridTemplateRows: `repeat(${rowsNb}, 12vw`
            }
        }

        document.documentElement.style.setProperty(`--board-${this.props.position}-width`, columnHeight);

        return (
            <div ref={(ref) => this.boardRef = ref} className={"Board " + this.props.position}
                 style={boardStyle}>
                {
                    this.props.members.map((member) => {
                        let photo;
                        try {
                            photo = require(`../../__mocks__/profilePicture/${member.photo}`);
                        } catch (e) {
                            photo = member.gender.label === 'F' ? defaultWoman : defaultMan;
                        }

                        return (
                            <div key={member.id}>
                                <img src={photo} alt=""/>
                                <p>
                                    {member.firstName} {member.lastName}<br/>
                                    {member.positions[0].label}
                                </p>
                            </div>
                        )
                    })
                }
            </div>
        );
    }
}

export default Board;
