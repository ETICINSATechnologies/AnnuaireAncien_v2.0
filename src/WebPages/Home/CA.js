import React, {Component} from 'react';
import defaultMan from '../../Images/default_man.svg'
import defaultWoman from '../../Images/default_woman.svg'

class CA extends Component {
    constructor(props) {
        super(props);
        this.state = {
            divHeight: 0
        };
        this.setDivHeight = this.setDivHeight.bind(this);
    }

    setDivHeight() {
        if (this.DOMBoard) {
            let divHeight = window.innerHeight - this.DOMBoard.getBoundingClientRect().y;
            this.setState({divHeight: divHeight});
        }
    }

    componentDidMount() {
        this.setDivHeight();
        window.addEventListener('resize', this.setDivHeight)
    }

    render() {
        const rowsNb = Math.max(this.props.members.length, 5);
        const rowSize = Math.min(this.state.divHeight / rowsNb, 250);

        let columnHeight;
        let boardStyle;

        if (window.innerHeight < window.innerWidth) {
            columnHeight = `${1.2 * rowSize}px`;
            boardStyle = {
                gridTemplateRows: `repeat(${rowsNb}, ${rowSize}px)`
            }
        }
        else {
            columnHeight = '12vw';
            boardStyle = {
                gridTemplateRows: `repeat(${rowsNb}, 12vw`
            }
        }

        document.documentElement.style.setProperty(`--board-${this.props.position}-width`, columnHeight);

        return (
            <div ref={(DOMBoard) => this.DOMBoard = DOMBoard} className={"CA " + this.props.position}
                 style={boardStyle}>
                {
                    this.props.members.map((member, index) => {
                        let photo;
                        try {
                            photo = require(`../../Tools/profilePicture/${member.photo}`);
                        } catch (e) {
                            photo = member.gender.label === 'F' ? defaultWoman : defaultMan;
                        }

                        return <div key={index} id={member.id}>
                            <img src={photo} alt=""/>
                            <p>
                                {member.firstName} {member.lastName}<br/>
                                {member.positions[0].label}
                            </p>
                        </div>
                    })
                }
            </div>
        );
    }
}

export default CA;
