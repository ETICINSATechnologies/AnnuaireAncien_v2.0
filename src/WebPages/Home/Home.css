.Home {
    display: grid;
    width: 100%;
    height: 100%;
    text-align: center;
}

.Home .home {
    display: grid;
    grid-template-columns: 12vw 1fr 12vw;
    grid-template-rows: 1fr;
    grid-gap: 1%;

    background-size: 100% auto;

    text-align: center;
    width: 100%;
    height: 100%;
}

.Home .home_title {
    vertical-align: middle;

    margin: auto;

    font-size: 5vw;
    width: 80%;
    height: 100%;
}

.Home .welcome {
    margin: 6vmin auto 10vmin auto;
}

.Home .welcome_etic {
    font-weight: bold;

    margin: auto;
}

.Home .left {
    background: rgba(0, 83, 96, 0.8);
}

.Home .right {
    background: rgba(155, 202, 139, 1);
}
