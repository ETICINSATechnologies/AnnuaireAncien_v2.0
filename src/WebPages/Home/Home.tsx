import React, {Component} from 'react';
import './Home.css';

import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";
import Board from "./Board";

import {Member, MemberInterface} from "../../Model/Member";

import {fetch} from "../../__mocks__/function/fetch";


type HomeState = {
    leftBoard: Member[]
    rightBoard: Member[]
}

class Home extends Component<{}, HomeState> {
    state = {
        leftBoard: [],
        rightBoard: []
    };

    componentDidMount() {
        fetch('api/v1/core/member/board/latest', {
            headers: {
                Authorization: Auth.getToken()
            }
        })
            .then(res => res.json())
            .then((result) => {
                // create an array of member from result.content
                let members = result.content.map((member: MemberInterface) => {
                    return new Member(member);
                }) as Member[];

                // split members in two parts, splice transforms the object on which it is called 
                let rightBoard = Array.from(members) as Member[];
                let leftBoard = rightBoard.splice(0, rightBoard.length / 2) as Member[];

                this.setState({
                    leftBoard: leftBoard,
                    rightBoard: rightBoard
                });
            });

        /*fetch('api/v1/core/member')
            .then((res: any) => res.json())
            .then((response: Page<Member>) => {
                console.log(response);
                let member = new Member(response.content[0]);
                console.log(member);
                console.log(member.read());
                member.department = {
                    id: 1,
                    label: 'kdlsdk',
                    name: ''
                };
                member.telephone = '01234567890';
                let m = member.update();
                //console.log(b.toString());
                fetch('api/v1/core/department', {
                    headers: {
                        Authorization: token
                    }
                })
                    .then(res => res.json())
                    .then((response) => {
                        let a = response as Department[];
                        console.log(a);
                        member.department = a[1];
                        console.log(JSON.stringify(m.positions[0].year));
                        fetch('api/v1/core/member/1', {
                            method: 'PUT',
                            headers: {
                                'Content-Type': 'application/json',
                                Authorization: token
                            },
                            body: JSON.stringify(m)
                        })
                            .then(res => res.json())
                            .then()
                    })
            })*/
    }

    render() {
        let activeButton = [];
        if (Auth.isConnected()) {
            activeButton.push('search');
            activeButton.push('profile');
        }
        activeButton = Auth.addCorrectButton(activeButton);

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <div className="Home">
                    <section className="home">
                        <Board members={this.state.leftBoard} position="left"/>
                        <div className="home_title">
                            <p className="welcome"> Bienvenue sur l'annuaire des anciens </p>
                            <p className="welcome_etic"> ETIC INSA TECHNOLOGIES </p>
                        </div>
                        <Board members={this.state.rightBoard} position="right"/>
                    </section>
                </div>
            </React.Fragment>
        );
    }
}

export default Home;
