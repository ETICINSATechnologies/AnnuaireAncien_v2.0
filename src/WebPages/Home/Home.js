import React, {Component} from 'react';
import './Home.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import CA from "./CA";
import Auth from "../../Components/Auth/Auth";
import fetch from "../../__mocks__/fetch";

class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            leftBoard: [],
            rightBoard: []
        }
    }

    componentDidMount() {
        fetch('api/v1/core/member/board/latest', {
            headers: {
                Authorization: Auth.getToken()
            }
        })
            .then(res => res.json())
            .then((result) => {
                if (result.content) {
                    let rightBoard = result.content;
                    let leftBoard = rightBoard.splice(0, rightBoard.length / 2);
                    this.setState({
                        leftBoard: leftBoard,
                        rightBoard: rightBoard
                    });
                }
            })
    }

    render() {
        console.log(this.state);
        let activeButton = [];
        if (Auth.isConnected()) {
            activeButton.push('search');
            activeButton.push('profile');
        }
        activeButton = Auth.addCorrectButton(activeButton);

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <div className="Home">
                    <section className="home">
                        <CA className="left_CA" members={this.state.leftBoard} position="left"/>
                        <div className="home_title">
                            <p className="welcome"> Bienvenue sur l'annuaire des anciens </p>
                            <p className="welcome_etic"> ETIC INSA TECHNOLOGIES </p>
                        </div>
                        <CA className="right_CA" members={this.state.rightBoard} position="right"/>
                    </section>
                </div>
            </React.Fragment>
        );
    }
}

export default Home;
