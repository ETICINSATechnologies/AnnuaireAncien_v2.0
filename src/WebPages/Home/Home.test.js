import React from 'react';
import ReactDOM from 'react-dom';
import Home from './Home';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from '../../Components/Auth/Auth';

Enzyme.configure({adapter: new Adapter()});

describe('Home unit tests', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Home/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Home unit test', () => {
        expect(
            Enzyme.shallow(<Home/>)
            .find('div.Home')
            .length
        ).toBe(1)
    });

    it('Home not connected snapshot test', () => {
        Auth.isConnected = jest.fn(() => false);

        const component = renderer.create(<Home/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('Home connected snapshot test', () => {
        Auth.isConnected = jest.fn(() => true);

        const component = renderer.create(<Home/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    })
});


