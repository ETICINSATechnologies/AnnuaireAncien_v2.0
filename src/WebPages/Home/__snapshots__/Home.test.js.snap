// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home unit tests Home connected snapshot test 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens 
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/search"
        onClick={[Function]}
      >
        rechercher
      </a>
      <a
        className="button"
        href="/profile"
        onClick={[Function]}
      >
        mon profil
      </a>
      <a
        className="button"
        href="/"
        onClick={[Function]}
      >
        se d√©connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
      onClick={[Function]}
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Home"
  >
    <section
      className="home"
    >
      <div
        className="Board left"
        style={
          Object {
            "gridTemplateRows": "repeat(5, 0px)",
          }
        }
      />
      <div
        className="home_title"
      >
        <p
          className="welcome"
        >
           Bienvenue sur l'annuaire des anciens 
        </p>
        <p
          className="welcome_etic"
        >
           ETIC INSA TECHNOLOGIES 
        </p>
      </div>
      <div
        className="Board right"
        style={
          Object {
            "gridTemplateRows": "repeat(5, 0px)",
          }
        }
      />
    </section>
  </div>,
]
`;

exports[`Home unit tests Home not connected snapshot test 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens 
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/connection"
        onClick={[Function]}
      >
        se connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
      onClick={[Function]}
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <div
    className="Home"
  >
    <section
      className="home"
    >
      <div
        className="Board left"
        style={
          Object {
            "gridTemplateRows": "repeat(5, 0px)",
          }
        }
      />
      <div
        className="home_title"
      >
        <p
          className="welcome"
        >
           Bienvenue sur l'annuaire des anciens 
        </p>
        <p
          className="welcome_etic"
        >
           ETIC INSA TECHNOLOGIES 
        </p>
      </div>
      <div
        className="Board right"
        style={
          Object {
            "gridTemplateRows": "repeat(5, 0px)",
          }
        }
      />
    </section>
  </div>,
]
`;
